### PR TITLE
🔪 Add text_slice function

### PR DESCRIPTION
**Pull request by nicpottier**
for https://github.com/nyaruka/goflow/issues/642

### CODE REVIEW

File: excellent/functions/builtin.go [hunk 1]
```diff
@@ -675,6 +676,48 @@ func Clean(env utils.Environment, text types.XText) types.XValue {
 	return types.NewXText(nonPrintableRegex.ReplaceAllString(text.Native(), ""))
 }
 
+// TextSlice returns the portion of `text` between `start` and `end` (inclusive).
```
Comment (rowanseymour) [inline excellent/functions/builtin.go]: start is inclusive, end is exclusive, right?

Comment (nicpottier) [reply excellent/functions/builtin.go]: yep yep, updated.

**Comment by nicpottier** [comment]
removed `left` and `right` and added migrations, let me know if that looks right @rowanseymour

File: legacy/expressions/testdata/legacy_tests.json [hunk 2]
```diff
@@ -649,17 +649,15 @@
         "errors": []
     },
     {
-        "template": "@(LEFT(\"abcdef\", 0))|@(LEFT(\"abcdef\", 2))|@(LEFT(\"واحد\", 2))|@(LEFT(\"err\", -1))",
+        "template": "@(LEFT(\"abcdef\", 0))|@(LEFT(\"abcdef\", 2))|@(LEFT(\"واحد\", 2))|@(LEFT(\"abcdef\", -1))",
```
Comment (nicpottier) [inline legacy/expressions/testdata/legacy_tests.json]: This is different behavior than before but I think I'm ok with it.

**Pull request merged**